tests/libtransmission/handshake-test.cc

```diff
@@ -330,24 +332,42 @@ TEST_F(HandshakeTest, incomingEncrypted)
 
     auto [io, sock] = createIncomingIo(session_);
 
+    auto res = std::optional<tr_handshake::Result>{};
+    auto const handshake = startHandshake(res, &mediator, io);
+
     // Peer->Client data from a successful encrypted handshake recorded
-    // in the wild for replay here
+    // in the wild for replay here. This test will play as the peer.
+    // 1. Peer sends Ya.
     sendB64ToClient(
         sock,
         "svkySIFcCsrDTeHjPt516UFbsoR+5vfbe5/m6stE7u5JLZ10kJ19NmP64E10qI"
         "nn78sCrJgjw1yEHHwrzOcKiRlYvcMotzJMe+SjrFUnaw3KBfn2bcKBhxb/sfM9"
         "J7nJ"sv);
+
+    // 2. Wait for client to reply with Yb.
+    ASSERT_TRUE(waitFor(
+        [&s = std::as_const(sock), buf = std::array<char, 16>{}, n_read = size_t{}]() mutable
+        {
+            if (auto ret = read(s, std::data(buf), std::size(buf)); ret > 0)
```

Comment:
Looked into why the Windows tests are failing.

Somehow this line is triggering an assert failure in the `read()` call...

```
minkernel\crts\ucrt\src\appcrt\lowio\read.cpp(380) : Assertion failed: fh >= 0 && (unsigned)fh < (unsigned)_nhandle
```